File: atcoder/abc367a_shout_everyday.hs
-- vicfred
-- https://atcoder.jp/contests/abc367/tasks/abc367_a
-- implementation
import Data.Bool (bool)

main :: IO ()
main = getLine >>= putStrLn . bool "No" "Yes" . f . map (read @Int) . words

f :: [Int] -> Bool
f [a, b, c] =
  if b < c then
    if b < a && a < c then
      False
    else
      True
  else
    if c < a && a < b then
      True
    else
      False

File: atcoder/abc367a_shout_everyday.cpp
// vicfred
// https://atcoder.jp/contests/abc367/tasks/abc367_a
// implementation
#include <iostream>

using namespace std;

int main() {
  int A, B, C;
  cin >> A >> B >> C;
  if (B > C) {
    if(A > C and A < B) {
      cout << "Yes" << endl;
    } else {
      cout << "No" << endl;
    }
    return 0;
  }
  if (A < B or A >= C) {
    cout << "Yes" << endl;
  } else {
    cout << "No" << endl;
  }
  return 0;
}
